qcodes/instrument_drivers/ZI/ZIUHFLI.py

```diff
@@ -2265,7 +2278,7 @@ def getduration():
             returndict =  self.scope.get(querystr)
             # The dict may have different 'depths' depending on the parameter.
             # The depth is encoded in the setting string (number of '/')
-            keys = setting.split('/')[1:]
```

Comment:
Is this change the change responsible for making the getters for `scope_mode` and `scope_average_weight` work?

Reply:
yes

Reply:
as the `setting` for `scope_mode` is just `mode` and for `scope_average_weight` is `averager/weight` then omitting the first part before a `/` of the setting leaves an empty list for the `mode` and a key error for the `weight`

Reply:
I see. And the `_scope_getter` is only used for the `scope_mode`, `scope_average_weight`, and `scope_duration` parameters, the latter of which is handled explicitly by the `_scope_getter` function.

The changes look correct. I just wonder why the `[1:]` was put in to begin with. Perhaps the setting nesting level changed in some firmware upgrade? If you can somehow check that, that would be fantastic. But since these changes make your version work, let's get them in.

Reply:
I went over the commit history of the original PR, and it seems that in commit `1066df6b01d6016da8ff7f34ccdfe201aea0fcfd` when the special case for `duration` was added this bug was introduced. Before that commit, `rawvalue = returndict` was outside and after the while loop `while keys != []` and then it should have worked.

Reply:
Wow, good job! Thank you.